pkg/build/model.go

```diff
@@ -0,0 +1,419 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/cache"
+	"github.com/okteto/okteto/pkg/env"
+	"github.com/okteto/okteto/pkg/filesystem"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model/utils"
+	"github.com/spf13/afero"
+)
+
+// BuildInfo represents the build info to generate an image
+type BuildInfo struct {
+	Secrets          BuildSecrets      `yaml:"secrets,omitempty"`
+	Name             string            `yaml:"name,omitempty"`
+	Context          string            `yaml:"context,omitempty"`
+	Dockerfile       string            `yaml:"dockerfile,omitempty"`
+	Target           string            `yaml:"target,omitempty"`
+	Image            string            `yaml:"image,omitempty"`
+	CacheFrom        cache.CacheFrom   `yaml:"cache_from,omitempty"`
+	Args             BuildArgs         `yaml:"args,omitempty"`
+	VolumesToInclude []VolumeMounts    `yaml:"-"`
+	ExportCache      cache.ExportCache `yaml:"export_cache,omitempty"`
+	DependsOn        BuildDependsOn    `yaml:"depends_on,omitempty"`
+}
+
+type VolumeMounts struct {
+	LocalPath  string
+	RemotePath string
+}
+
+// BuildArg is an argument used on the build step.
+type BuildArg struct {
+	Name  string
+	Value string
+}
+
+func (v *BuildArg) String() string {
+	value, err := env.ExpandEnv(v.Value)
+	if err != nil {
+		return fmt.Sprintf("%s=%s", v.Name, v.Value)
+	}
+	return fmt.Sprintf("%s=%s", v.Name, value)
+}
+
+// BuildArgs is a list of arguments used on the build step.
+type BuildArgs []BuildArg
+
+// BuildDependsOn represents the images that needs to be built before
+type BuildDependsOn []string
+
+// BuildSecrets represents the secrets to be injected to the build of the image
+type BuildSecrets map[string]string
+
+// ManifestBuild defines all the build section
+type ManifestBuild map[string]*BuildInfo
+
+// GetDockerfilePath returns the path to the Dockerfile
+func (b *BuildInfo) GetDockerfilePath() string {
+	if filepath.IsAbs(b.Dockerfile) {
+		return b.Dockerfile
+	}
+	fs := afero.NewOsFs()
```

Comment:
Given that this is a refactor. Should we move the fs as parameter or extract it on a new method that already is initialised. This way this function could be tested properly. WDYT?